src/main/scala/viper/silver/parser/Resolver.scala

```diff
@@ -266,12 +212,9 @@ case class TypeChecker(names: NameAnalyser) {
         check(cond, Bool)
         invs foreach (check(_, Bool))
         check(body)
-      case PLocalVarDecl(_, typ, init) =>
-        check(typ)
-        init match {
-          case Some(i) => check(i, typ)
-          case None =>
-        }
+      case PVars(vars, initial) =>
+        vars foreach (v => check(v.typ))
+        initial.map(i => check(PAssign(vars.map(v => PIdnUse(v.idndef.name)(v.idndef.pos)), i)(i.pos)))
```

Comment:
I guess you could write checkAssign s.t. it takes Seq[PIdnUse], PExp etc. so that you don't have to create a ""fake"" statement here. But I'm aware I'm nitpicking now, this is also fine ^^